test/integration-test/src/tests/lsm.rs

```diff
@@ -0,0 +1,54 @@
+use std::{fs::File, io::{ErrorKind, Write}, path::Path};
+use aya::{programs::Lsm, util::KernelVersion, Btf, Ebpf};
+use std::net::{Ipv4Addr, SocketAddr, SocketAddrV4, TcpListener};
+use nix::{
+    sys::wait::waitpid,
+    unistd::{fork, getpid, ForkResult},
+};
+
+#[test]
+fn lsm_cgroup() {
+    let kernel_version = KernelVersion::current().unwrap();
+    if kernel_version < KernelVersion::new(6, 0, 0) {
+        eprintln!("skipping lsm_cgroup test on kernel {kernel_version:?}");
+        return;
+    }
+
+    let mut bpf: Ebpf = Ebpf::load(crate::TEST).unwrap();
+    let prog: &mut Lsm = bpf.program_mut("test_lsmcgroup").unwrap().try_into().unwrap();
+    let btf = Btf::from_sys_fs().expect("could not get btf from sys");
+    if let Err(err) = prog.load("socket_bind", &btf) {
+        panic!("{err}");
+    }
+
+    let cgroup_path = Path::new(".").join("/sys/fs/cgroup/").join("lsm_cgroup_test");
```

Comment:
These multiple `join`s look unnecessary.

```suggestion
    let cgroup_path = Path::new("/sys/fs/cgroup/lsm_cgroup_test");
```